orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -377,7 +399,11 @@ public void initServiceRun(List<GatewayConfig> allGateway, Set<Node> allNodes, S
             }
             grainUploader.uploadGrains(allNodes, saltConfig.getGrainsProperties(), exitModel, sc, exitCriteria);
 
+            // TODO Question: Is SyncAllRunner needed? This takes about 1.8s. Operations seem to work fine without this.
+            //  Maybe this call is more optimal when adding a large set of nodes?
             saltCommandRunner.runSaltCommand(sc, new SyncAllRunner(allNodeHostname, allNodes), exitModel, exitCriteria);
+
+            // TODO Question: Similarly for MineUpdateRunner. About 1 second. Are mine updates required?
```

Comment:
Mine is used as a cache and it's used for unbound configuration which is important during upscale.